external/fv3fit/tests/reservoir/test_readout.py

```diff
@@ -27,48 +28,113 @@ def test_square_even_terms(arr, axis, expected):
+        intercepts=np.zeros(2),
     )
     input = np.array([2, 2, 2, 2, 2])
     # square even entries -> [4, 2, 4, 2, 4]
     output = readout.predict(input)
-    np.testing.assert_array_almost_equal(output, np.array([16.0 / 5, 16.0 / 5]))
+    np.testing.assert_array_almost_equal(output, np.array([16.0, 16.0]))
 
 
 def test_reservoir_computing_readout_fit():
-    unsquared = ReservoirComputingReadout(Ridge())
-    squared = ReservoirComputingReadout(Ridge(), square_half_hidden_state=True)
+    unsquared = ReservoirComputingReadout(
+        ReadoutHyperparameters(linear_regressor_kwargs={})
+    )
+    squared = ReservoirComputingReadout(
+        ReadoutHyperparameters(
+            linear_regressor_kwargs={}, square_half_hidden_state=True
+        )
+    )
 
     res_states = np.arange(10).reshape(2, 5)
     output_states = np.arange(6).reshape(2, 3)
 
     unsquared.fit(res_states, output_states)
     squared.fit(res_states, output_states)
 
-    assert not np.allclose(
-        unsquared.linear_regressor.coef_, squared.linear_regressor.coef_
+    assert not np.allclose(unsquared.coefficients, squared.coefficients)
+
+
+def test_combined_readout():
+    state_size = 3
+    output_size = 2
+    readout_1 = ReservoirComputingReadout(
+        hyperparameters=ReadoutHyperparameters(
+            linear_regressor_kwargs={}, square_half_hidden_state=False
+        ),
+        coefficients=np.ones(shape=(output_size, state_size)),
+        intercepts=np.zeros(output_size),
+    )
+    readout_2 = ReservoirComputingReadout(
+        hyperparameters=ReadoutHyperparameters(
+            linear_regressor_kwargs={}, square_half_hidden_state=False
+        ),
+        coefficients=2.0 * np.ones(shape=(output_size, state_size)),
+        intercepts=np.zeros(output_size),
+    )
+    input = np.array([1, 1, 1])
+    output_1 = readout_1.predict(input)
+    output_2 = readout_2.predict(input)
+
+    combined_readout = CombinedReservoirComputingReadout(
+        readouts=[readout_1, readout_2]
+    )
+    combined_input = np.concatenate([input, input])
+    output_combined = combined_readout.predict(combined_input)
+
+    np.testing.assert_array_almost_equal(
+        output_combined, np.concatenate([output_1, output_2])
+    )
+
+
+def test_combined_readout_inconsistent_hyperparameters():
+    readout_1 = ReservoirComputingReadout(
+        hyperparameters=ReadoutHyperparameters(
+            linear_regressor_kwargs={}, square_half_hidden_state=True
+        ),
+        coefficients=np.ones(shape=(2, 2)),
+        intercepts=np.zeros(2),
+    )
+    readout_2 = ReservoirComputingReadout(
+        hyperparameters=ReadoutHyperparameters(
+            linear_regressor_kwargs={}, square_half_hidden_state=False
+        ),
+        coefficients=np.ones(shape=(2, 2)),
+        intercepts=np.zeros(2),
+    )
+    with pytest.raises(ValueError):
+        CombinedReservoirComputingReadout(readouts=[readout_1, readout_2])
+
+
+def test_combined_load(tmpdir):
```

Comment:
Can this test be simplified by writing a `CombinedReservoirComputingReadout.dump` method, that automatically determines subdirectory names? Do we expect the ability to combine paths from potentially separate training runs to be used?

Reply:
> Do we expect the ability to combine paths from potentially separate training runs to be used?
No, I don't see that happening.

Since training will be in parallel, I think it will be easier to have the training function handle saving all the separate readouts into subdirectories rather than waiting for all the readouts to train before combining into that class to dump.